### PR TITLE
CU-3bfmmke - XCVM Gateway

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**

### CODE REVIEW

File: code/Cargo.toml [hunk 1]
```diff
@@ -68,6 +69,7 @@ opt-level = "s"
 overflow-checks = true
 panic = "abort"
 rpath = false
+codegen-units = 1
```
Comment (KaiserKarel) [inline code/Cargo.toml:72]: Shouldn't this be set through LTO?

File: code/xcvm/SPEC.md [hunk 2]
```diff
@@ -330,24 +330,60 @@ Indices in bindings must to be **sorted** in an ascending order and **unique**.
...
+
+### 2.2.3.1.2. Spawn receive
+
+Upon reception of a `SpawnPackage`, the XCVM execution MUST happen on a
+sub-transaction (the transaction MUST not fail even if the execution fails) and
+an XCVM-specific acknowledgement must be committed for the packet:
+- A single byte, `0x00` if unsuccessful
+- A single byte, `0x01` if successful
+
+The bridge MUST mint the **assets** in the Router contract before executing the
```
Comment (KaiserKarel) [inline code/xcvm/SPEC.md:379]:
```suggestion
The bridge MUST deposit the **assets** in the Router contract before executing the
```

**Review by KaiserKarel — APPROVED** [review]

**Review by aeryz — APPROVED** [review]

**Review by cor — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]
Reviewed dic and toml and lock files as @nix

**Comment by ComposableFiB0t** [comment]
Task linked: [XCVM-229 XCVM IBC Gateway contract](https://app.clickup.com/t/20465559/XCVM-229)

**Pull request merged**